consensus/clique/src/main/java/tech/pegasys/pantheon/consensus/clique/CliqueBlockHashing.java

```diff
@@ -48,6 +47,9 @@ public static Hash calculateDataHashForProposerSeal(
    */
   public static Address recoverProposerAddress(
       final BlockHeader header, final CliqueExtraData cliqueExtraData) {
+    if (header.getNumber() == BlockHeader.GENESIS_BLOCK_NUMBER) {
```

Comment:
nit: Personally, I'd move this inside the "isPresent" check - just to minimise the conditionals executed on every check ... its probably a minor benefit at best.

Reply:
done